testplan/testing/multitest/driver/base.py

```diff
@@ -337,34 +348,36 @@ def _setup_file_logger(self, path):
         When a file logger is finished with, _close_file_logger() should be
         called to close the opened file object and release the file handle.
         """
-        if self.file_logger is not None:
-            raise RuntimeError("File logger already exists")
-
-        formatter = logging.Formatter("%(asctime)s %(levelname)s %(message)s")
-        handler = logging.FileHandler(path)
-        handler.setFormatter(formatter)
-        logger = logging.getLogger("FileLogger_{}".format(self.cfg.name))
-        logger.addHandler(handler)
-        logger.setLevel(self.logger.getEffectiveLevel())
-        self.file_logger = logger
-        self.file_logger.propagate = False  # No console logs
+        if self._file_log_handler is not None:
+            raise self.logger.debug(
+                "{}: File logger already exists".format(self)
+            )
+
+        # Note that in unit test driver's runpath might not be set
+        if self.cfg.file_logger and self.runpath is not None:
+            formatter = logging.Formatter(
+                "%(asctime)s %(levelname)s %(message)s"
+            )
+            self._file_log_handler = logging.FileHandler(
+                os.path.join(self.runpath, self.cfg.file_logger)
+            )
+            self._file_log_handler.setFormatter(formatter)
+            self.logger.addHandler(self._file_log_handler)
+            self.logger.propagate = False  # No console logs
 
     def _close_file_logger(self):
         """
-        Closes a logfile previously opened by _setup_file_logger() and removes
-        the logger from self.file_logger. This should be called when the file
+        Closes a handler previously opened by _setup_file_logger() and removes
+        the file handler from self.logger. This should be called when the file
         logger is done with to avoid leaking file handles - typically this
         should be called from stopping().
         """
-        if self.file_logger is None:
```

Comment:
Why are we moving this check?

Reply:
this `self.file_logger` is the same thing as `self.logger` and it has a handler to write to file. now the handler is `self._file_log_handler`. if `self.cfg.file_logger` is set then `self._file_log_handler` is not None, we'll close the handler.